kv/kvserver/concurrency: TestConcurrencyManagerBasic failed

[(kv/kvserver/concurrency).TestConcurrencyManagerBasic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2483597&tab=buildLog) on [master@6980223544fff24f0d2636fccad314e8c7511073](https://github.com/cockroachdb/cockroach/commits/6980223544fff24f0d2636fccad314e8c7511073):

```
=== RUN   TestConcurrencyManagerBasic
    test_log_scope.go:150: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestConcurrencyManagerBasic888195078
    test_log_scope.go:61: use -show-logs to present logs inline
--- FAIL: TestConcurrencyManagerBasic (2.39s)
=== RUN   TestConcurrencyManagerBasic/deadlocks
    --- FAIL: TestConcurrencyManagerBasic/deadlocks (0.32s)
=== RUN   TestConcurrencyManagerBasic/deadlocks/lock_ordering_complex
    datadriven.go:243: 
        testdata/concurrency_manager/deadlocks:389: 
        expected:
        [-] finish req4w: finishing request
        [7] sequence req3w2: acquiring latches
        [7] sequence req3w2: scanning lock table for conflicting locks
        [7] sequence req3w2: sequencing complete, returned guard
        
        found:
        [-] finish req4w: finishing request
        [7] sequence req3w2: acquiring latches
        [7] sequence req3w2: scanning lock table for conflicting locks
        [7] sequence req3w2: sequencing complete, returned guard
        [7] sequence req3w2: blocked on select in concurrency_test.(*cluster).PushTransaction
        --- FAIL: TestConcurrencyManagerBasic/deadlocks/lock_ordering_complex (0.18s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestConcurrencyManagerBasic PKG=./pkg/kv/kvserver/concurrency TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestConcurrencyManagerBasic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/kv/kvserver/concurrency/concurrency_manager_test.go
+++ b/pkg/kv/kvserver/concurrency/concurrency_manager_test.go
@@ -985,7 +985,7 @@ func (m *monitor) waitForAsyncGoroutinesToStall(t *testing.T) {
 		// Add a small fixed delay after each iteration. This is sufficient to
 		// prevents false detection of stalls in a few cases, like when
 		// receiving on a buffered channel that already has an element in it.
-		defer time.Sleep(1 * time.Millisecond)
+		defer time.Sleep(5 * time.Millisecond)
 
 		prevStatus := status
 		status = goroutineStatus(t, filter, &m.buf)

